cli/lsp/diagnostics.rs

```diff
@@ -847,8 +853,20 @@ fn diagnose_resolved(
               .push(DenoDiagnostic::NoAssertType.to_lsp_diagnostic(&range)),
           }
         }
-      } else if NpmPackageReference::from_specifier(specifier).is_ok() {
-        // ignore npm specifiers for now
+      } else if let Ok(pkg_ref) = NpmPackageReference::from_specifier(specifier)
+      {
+        if let Some(npm_resolver) = &snapshot.maybe_npm_resolver {
+          // show diagnostics for npm package references that aren't cached
+          if npm_resolver
+            .resolve_package_folder_from_deno_module(&pkg_ref.req)
+            .is_err()
```

Comment:
That this handle situation when user has `--node-modules-dir` flag specified?

Reply:
Nope. That's not supported at the moment because there's no way to provide that to the lsp at the moment (see second point in pr description)